packages/overlay/src/overlay-stack.ts

```diff
@@ -187,33 +188,21 @@ export class OverlayStack {
     }
 
     private addEventListeners(): void {
+        this._eventsAreBound = true;
```

Comment:
Can we move the `_eventsAreBound` check here so that callers of `addEventListeners` don't need to be aware of it?

```ts
if (this._eventsAreBound) {
  return;
}
this._eventsAreBound = true;
```

Then `openOverlay` could safely call `this.addEventListeners()` directly.